lib/api/src/sys/externals/function.rs

```diff
@@ -401,24 +430,61 @@ impl Function {
             *slot = arg.as_raw(store);
         }
 
+        // Invoke the call
+        self.call_wasm_raw(store, trampoline, values_vec, results)?;
+        Ok(())
+    }
+
+    #[cfg(feature = "compiler")]
+    fn call_wasm_raw(
+        &self,
+        store: &mut impl AsStoreMut,
+        trampoline: VMTrampoline,
+        mut params: Vec<RawValue>,
+        results: &mut [Value],
+    ) -> Result<(), RuntimeError> {
         // Call the trampoline.
-        let vm_function = self.handle.get(store.as_store_ref().objects());
-        if let Err(error) = unsafe {
-            wasmer_call_trampoline(
-                store.as_store_ref().signal_handler(),
-                vm_function.anyfunc.as_ptr().as_ref().vmctx,
-                trampoline,
-                vm_function.anyfunc.as_ptr().as_ref().func_ptr,
-                values_vec.as_mut_ptr() as *mut u8,
-            )
-        } {
```

Comment:
Can we add a new comment linking to the asyncify refactor afterwards:

```
// TODO: issue #XYZ 
```

Reply:
done